src/Faker/Container/ContainerBuilder.php

```diff
@@ -38,19 +38,6 @@ public function add($value, string $name = null): self
             ));
         }
 
-        if ($name === null) {
-            if (is_string($value)) {
-                $name = $value;
-            } elseif (is_object($value)) {
-                $name = get_class($value);
-            } else {
-                throw new \InvalidArgumentException(sprintf(
-                    'Second argument to "%s::add()" is required not passing a string or object as first argument',
-                    self::class,
-                ));
-            }
-        }
```

Comment:
By requiring to pass `$name` to this method, we can get rid of this magic - it's better to be explicit.